docs/kms/README.md

```diff
@@ -1,56 +1,139 @@
-# KMS Quickstart Guide [![Slack](https://slack.min.io/slack?type=svg)](https://slack.min.io)
+# KMS Guide [![Slack](https://slack.min.io/slack?type=svg)](https://slack.min.io)
 
 MinIO uses a key-management-system (KMS) to support SSE-S3. If a client requests SSE-S3, or auto-encryption
 is enabled, the MinIO server encrypts each object with an unique object key which is protected by a master key
-managed by the KMS. Usually all object keys are protected by a single master key.
+managed by the KMS.
 
-MinIO supports two different KMS concepts:
- - External KMS:
-   MinIO can be configured to use an external KMS i.e. [Hashicorp Vault](https://www.vaultproject.io/).
-   An external KMS decouples MinIO as storage system from key-management. An external KMS can
-   be managed by a dedicated security team and allows you to grant/deny access to (certain) objects
-   by enabling or disabling the corresponding master keys on demand.
+> MinIO still provides native Hashicorp Vault support. However, this is feature is **deprecated** and may be
+> removed in the future. Therefore, we strongly recommend to use the architecture and KMS Guide below.
+> If you have to maintain a legacy MinIO-Vault deployment you can find the legacy documentation [here](./vault-legacy.md).
 
-- Direct KMS master keys:
-   MinIO can also be configured to directly use a master key specified by the environment variable `MINIO_KMS_MASTER_KEY` or with a docker secret key.
-   Direct master keys are useful if the storage backend is not on the same machine as the MinIO server, e.g.,
-   if network drives or MinIO gateway is used and an external KMS would cause too much management overhead.
+## Architecture and Concepts
 
-   Note: KMS master keys are mainly for testing purposes. It's not recommended to use them for production deployments.
-   Further if the MinIO server machine is ever compromised, then the master key must also be treated as compromised.
+The KMS decouples MinIO as an application-facing storage system from the secure key storage and
+may be managed by a dedicated security team. In general, the MinIO-KMS infrastructure looks like this:
+```
+ +-------+       +-----+
```

Comment:
If you need help with proper pictures we should engage @belvid @labebe6 here.  I think we can avoid ASCII pictures here. 

This shouldn't stop the PR itself to be merged because its a great change nonetheless.